dockstore-webservice/src/main/java/io/dockstore/webservice/core/VersionMetadata.java

```diff
@@ -59,6 +60,7 @@
     protected String verifiedSource;
 
     @Column()
+    @Pattern(regexp = "10\\..+/.+")
```

Comment:
Just curious, if you remove the migrations code, does this cause a check migrations error? I suspect not.

Reply:
Also, belated thought/question... What does this actually do? Does the DropWizard only affect deserialization?

I ask because I use it for the Dockstore.yaml, but I have to explicitly invoke the validator as I recall.

I know we use this in a couple of other places, but I'm just curious what the practical effect of this is. Because we don't let you edit the version metadata directly, I'm not sure this will ever come into play.

Reply:
I believe Hibernate checks the `@Pattern` annotation before persisting, see this reference:

https://thorben-janssen.com/automatically-validate-entities-with-hibernate-validator/#Automatic_Validation_upon_Lifecycle_Events

Indeed, with both the `@Pattern` annotation and db constraint, when I `setDoiURL("illegal-value")`, the `@Pattern` annotation makes the webservice respond with a generic 400 "Bad Request".  If I do the same, but remove the `@Pattern` annotation, the webservice responds with a 409 that indicates the db constraint violated.

Since, in our use case, the `@Pattern` annotation and db constraint are more-or-less functionally redundant, I might remove the `@Pattern`, since in the current implementation, the db constraint produces a more useful error message.